test/e2e/federated-service.go

```diff
@@ -369,22 +453,24 @@ func discoverService(f *framework.Framework, name string, exists bool) {
 	}
 
 	_, err := f.Client.Pods(f.Namespace.Name).Create(pod)
-	Expect(err).NotTo(HaveOccurred(), "Trying to create pod to run %q", command)
-	defer f.Client.Pods(f.Namespace.Name).Delete(FederatedServicePod, api.NewDeleteOptions(0))
+	framework.ExpectNoError(err, "Trying to create pod to run %q", command)
+	defer func() { f.Client.Pods(f.Namespace.Name).Delete(podName, api.NewDeleteOptions(0)) }()
 
 	if exists {
 		// TODO(mml): Eventually check the IP address is correct, too.
-		Eventually(podExitCodeDetector(f, FederatedServicePod, 0), 10*DNSTTL, time.Second*2).
+		Eventually(podExitCodeDetector(f, podName, 0), 2*DNSTTL, time.Second*2).
```

Comment:
Aah - never mind.  It's the DNS caching issue, I guess.  This is trying to resolve the DNS name before the DNS records have propagated, and hence caching the CNAME result.  I should perhaps, in this test, wait until the DNS records have propagated before querying them, just to speed things up a bit.